Add docker registry, orchestrator and worker namespaces

chmeliik opened the pull request:
Also fixed some issues discovered in the process, commit messages should provide enough explanation.

The only thing that is missing now is the buildroot.

__TODO__: 
* [x] update README, but only minor changes until buildroot is done

chmeliik commented:
Removed deprecated items from client config secret

MartinBasti commented on cleanup.yaml:
```diff
@@ -99,6 +99,15 @@
       when: whoami.rc == 0
 
 
+# TODO
```
:eyes:

StykMartin commented on deploy.yaml:
```diff
@@ -237,3 +237,11 @@
           - "system:build-strategy-custom"
+
+
+# TODO
```
:eyes:

chmeliik commented:
Removed the `insecure` option for openshift from reactor-config-map, removed empty plays :eyes:

MartinBasti commented:
Works like charm, except I got this traceback when I run build.

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/koji/daemon.py", line 1292, in runTask
    response = (handler.run(),)
  File "/usr/lib/python3.7/site-packages/koji/tasks.py", line 314, in run
    return koji.util.call_with_argcheck(self.handler, self.params, self.opts)
  File "/usr/lib/python3.7/site-packages/koji/util.py", line 263, in call_with_argcheck
    return func(*args, **kwargs)
  File "/usr/lib/koji-builder-plugins/builder_containerbuild.py", line 913, in handler
    triggered_after_koji_task=opts.get('triggered_after_koji_task', None),
  File "/usr/lib/koji-builder-plugins/builder_containerbuild.py", line 568, in runBuilds
    semi_results = self.createContainer(**kwargs)
  File "/usr/lib/koji-builder-plugins/builder_containerbuild.py", line 642, in createContainer
    build_response = create_method(**orchestrator_create_build_args)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 57, in catch_exceptions
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 882, in create_orchestrator_build
    return self._do_create_prod_build(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 784, in _do_create_prod_build
    response = self._create_build_config_and_build(build_request)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 473, in _create_build_config_and_build
    existing_bc = self.os.create_build_config(json.dumps(build_json)).json()
  File "/usr/local/lib/python3.7/site-packages/osbs/http.py", line 216, in json
    raise OsbsResponseException(text, self.status_code)
osbs.exceptions.OsbsResponseException: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the API version in the data (v1) does not match the expected API version (build.openshift.io/v1)","reason":"BadRequest","code":400}
```

This is caused by old metadata in `/usr/shares/osbs/orchestrator.json`. It seems that those tempaltes are not updated from git installation so `osbs-client` from master code fails.

MartinBasti commented:
I also noticed that rebuilds are done from cache, I'd prefer `no-cache` builds to avoid nasty race conditions when I want just update code from git

https://docs.openshift.com/container-platform/3.11/dev_guide/builds/build_strategies.html#no-cache

chmeliik commented:
> This is caused by old metadata in /usr/shares/osbs/orchestrator.json. It seems that those tempaltes are not updated from git installation so osbs-client from master code fails.

@MartinBasti the templates should be updated, but by default, pip3 will install everything in `/usr/local/`, so they get updated in the wrong location. This should not be the case for the koji images, because `koji-base` sets the `PIP_PREFIX=/usr` environment variable.

I suspect the traceback is not from `koji-builder` but rather the buildroot - speaking of, what are you using for that?

chmeliik commented:
Added `noCache` options for koji base, client, builder and hub (db should not need it), changed default names for namespace, increased PV "capacity" (it probably does not do anything)

MartinBasti commented:
Again, new origin installation with fetched PR changes

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/koji/daemon.py", line 1292, in runTask
    response = (handler.run(),)
  File "/usr/lib/python3.7/site-packages/koji/tasks.py", line 314, in run
    return koji.util.call_with_argcheck(self.handler, self.params, self.opts)
  File "/usr/lib/python3.7/site-packages/koji/util.py", line 263, in call_with_argcheck
    return func(*args, **kwargs)
  File "/usr/lib/koji-builder-plugins/builder_containerbuild.py", line 913, in handler
    triggered_after_koji_task=opts.get('triggered_after_koji_task', None),
  File "/usr/lib/koji-builder-plugins/builder_containerbuild.py", line 568, in runBuilds
    semi_results = self.createContainer(**kwargs)
  File "/usr/lib/koji-builder-plugins/builder_containerbuild.py", line 642, in createContainer
    build_response = create_method(**orchestrator_create_build_args)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 57, in catch_exceptions
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 882, in create_orchestrator_build
    return self._do_create_prod_build(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 784, in _do_create_prod_build
    response = self._create_build_config_and_build(build_request)
  File "/usr/local/lib/python3.7/site-packages/osbs/api.py", line 473, in _create_build_config_and_build
    existing_bc = self.os.create_build_config(json.dumps(build_json)).json()
  File "/usr/local/lib/python3.7/site-packages/osbs/http.py", line 216, in json
    raise OsbsResponseException(text, self.status_code)
osbs.exceptions.OsbsResponseException: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the API version in the data (v1) does not match the expected API version (build.openshift.io/v1)","reason":"BadRequest","code":400}
```

chmeliik commented:
Man, I thought docker was supposed to fix the "works on my machine" issue:

`$ oc -n osbs-koji rsh dc/koji-builder cat /usr/share/osbs/orchestrator.json`
```json
{
  "apiVersion": "build.openshift.io/v1",
  "kind": "BuildConfig",
  ...
}
```
`$ oc -n osbs-koji rsh dc/koji-builder ls /usr/local/lib/python3.7/`
```
ls: cannot access '/usr/local/lib/python3.7/': No such file or directory
command terminated with exit code 2
```

MartinBasti commented:
```
oc -n osbs-koji rsh dc/koji-builder cat /usr/share/osbs/orchestrator.json
{
  "apiVersion": "v1",
  "kind": "BuildConfig",
  "metadata": {
....
```

I'm not using overrides

chmeliik commented:
@MartinBasti 
> I'm not using overrides

Right, that means the images are built from `containerbuildsystem/osbs-box@a-new-box`. You need to override the defaults to make openshift build from this PR (either use my repo, or somehow specify this PR instead of the `a-new-box` branch if that is possible)
